pkg/controller/namespace/namespace_controller.go

```diff
@@ -175,7 +189,11 @@ func (nm *NamespaceController) syncNamespaceFromKey(key string) (err error) {
 		utilruntime.HandleError(fmt.Errorf("Unable to retrieve namespace %v from store: %v", key, err))
 		return err
 	}
-	return nm.namespacedResourcesDeleter.Delete(namespace.Name)
+	err = nm.namespacedResourcesDeleter.Delete(namespace.Name)
+	if err != nil && nm.recorder != nil {
+		nm.recorder.Event(namespace, corev1.EventTypeWarning, "DeletionFailed", err.Error())
```

Comment:
if the event is emitted in the namespace that is undergoing deletion, the event should not persist because create operations are blocked in a namespace marked for deletion during admission.

Reply:
I can unfortunately not tell you why, but events are always created in the `default` namespace and associated to the object they should report on via the `involvedObject` field. A sample event from a kube-controller-manager with this patch looks like this:

```
apiVersion: v1
items:
- apiVersion: v1
  count: 566
  eventTime: null
  firstTimestamp: "2019-08-04T17:27:30Z"
  involvedObject:
    apiVersion: v1
    kind: Namespace
    name: test
    resourceVersion: "66960469"
    uid: 4173c511-656d-40b2-8d3c-c45fc577efd0
  kind: Event
  lastTimestamp: "2019-08-06T15:17:30Z"
  message: 'unable to retrieve the complete list of server APIs: upload.cdi.kubevirt.io/v1alpha1:
    the server is currently unable to handle the request'
  metadata:
    creationTimestamp: "2019-08-04T17:27:30Z"
    name: test.15b7c8521b8aa8ea
    namespace: default
    resourceVersion: "67525313"
    selfLink: /api/v1/namespaces/default/events/test.15b7c8521b8aa8ea
    uid: 3ce6b2fb-9b8e-4bf7-ace5-f0d6fb9752e8
  reason: NamespaceDeletionFailed
  reportingComponent: ""
  reportingInstance: ""
  source:
    component: namespace_cleanup_controller
  type: Warning
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```